### ISSUE TITLE
Random forest probability output 

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#3850]
The current implementation only outputs binary labels:
https://gist.github.com/olinguyen/ade8e2bd0899b2ec9bee00d03fbb5172

It would be nicer if the forest would average the probabilistic predictions of each tree, as e.g. [done in sklearn](http://scikit-learn.org/stable/modules/ensemble.html#forest)

> In contrast to the original publication [B2001], the scikit-learn implementation combines classifiers by averaging their probabilistic prediction, instead of letting each classifier vote for a single class.

**Comment by vigsterkr** [comment]
@karlnapf for this would you use https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/ensemble/MeanRule.h
once RF can output probabilities for classification?

**Comment by karlnapf** [comment]
Yes in general an average is used. We will have to check the details though.

**Comment by karlnapf** [comment]
Done in #3954
